tests/mgr/test_dashboard_url.py

```diff
@@ -0,0 +1,77 @@
+from cli.cephadm.cephadm import CephAdm
+from cli.utilities.utils import get_service_id
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class DashboardUrlError(Exception):
+    pass
+
+
+class CephMgrServiceFailError(Exception):
+    pass
+
+
+class ResourceNotFoundError(Exception):
+    pass
+
+
+def check_dashboard_url(ceph_cluster):
+    """
+    Get dashboard URL from mgr service and check if dashboard URL does not
+    bind to host.containers.internal
+    """
+    url = ceph_cluster.get_mgr_services().get("dashboard")
+    if not url:
+        raise ResourceNotFoundError("Dashboard not configured")
+    if "host.containers.internal" in url:
+        log.error("Dashboard URL {} is not binding to IP of hostname")
+        return False
+    log.info("Dashboard URL is {}".format(url))
+    return True
+
+
+def run(ceph_cluster, **kw):
+    """
+    Check if the Ceph dashboard URL consists of host IP and
+    does not bind to host.containers.internal before and after
+    container restart and mgr failover is performed
+    """
+    # Get mgr node
+    node = ceph_cluster.get_nodes(role="mgr")[0]
+
+    # Check if dashboard URL does not bind to host.containers.internal
+    check_url = check_dashboard_url(ceph_cluster)
+    if check_url is False:
+        raise DashboardUrlError("Dashboard URL binds to host.containers.internal")
+    log.info("Dashboard URL does NOT bind to host.containers.internal")
+
+    # Get container id for mgr daemon
+    out, rc = node.exec_command(sudo=True, cmd="podman ps | grep mgr")
+    container_id = out.split(" ")[0]
+
+    # Restart container for mgr daemon
+    out, rc = node.exec_command(sudo=True, cmd=f"podman restart {container_id}")
```

Comment:
instead add this method here - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/utils.py#L100

Reply:
Done